pkg/util/fs.go

```diff
@@ -77,39 +75,66 @@ func (h *fs) Exists(file string) bool {
 	return err == nil
 }
 
-// Copy copies a set of files from sourcePath to targetPath
-func (h *fs) Copy(sourcePath string, targetPath string) error {
-	if _, err := os.Stat(sourcePath); err != nil {
+// Copy copies the source file to a destination directory. If the destination is
+// a filename, the file will be replaced.
+func (h *fs) Copy(source string, dest string) (err error) {
+	sourcefile, err := os.Open(source)
+	if err != nil {
 		return err
 	}
 
-	info, err := os.Stat(targetPath)
-
-	if err != nil || (info != nil && !info.IsDir()) {
-		err = os.Mkdir(targetPath, 0700)
-		if err != nil {
-			return err
-		}
+	defer sourcefile.Close()
+	glog.V(5).Infof("%q -> %q", source, dest)
+	destfile, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer destfile.Close()
 
-		targetPath = filepath.Join(targetPath, filepath.Base(sourcePath))
+	if _, err := io.Copy(destfile, sourcefile); err != nil {
+		return err
+	}
+	sourceinfo, err := os.Stat(source)
+	if err != nil {
+		return err
 	}
-	// TODO: Use the appropriate command for Windows
-	glog.V(5).Infof("cp -a %s %s", sourcePath, targetPath)
-	return h.runner.Run("cp", "-a", sourcePath, targetPath)
+	return os.Chmod(dest, sourceinfo.Mode())
```

Comment:
You can make the chmod conditional on os != "windows" for it to truly work on windows